### PR TITLE
New documentation using mkdocs

### PR DESCRIPTION
**Pull request by BastLeblanc**
*Issue #, if available:*

*Description of changes:*
- mkdocs.yml
- github action to deploy to github pages
- move documentation to ./docs
- create more pages to describe demos

*Description of testing performed to validate your changes (required if pull request includes CloudFormation or source code changes):*
deploy on the dev account.

When merging we need to update the https://docs.retaildemostore.retail.aws.dev DNS and github pages configuration

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/Technical Guides/EnablingShopperProfiles.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+
```
Comment (gfaires) [inline docs/Technical Guides/EnablingShopperProfiles.md:1]: General comment on the Technical Guides.  They seem to be copies of the Demo Guide that can be launched from within the web-ui.  Therefore, we need to maintain the information in two places.  Is it possible to pull this content from the demo guide?

File: docs/partner-integrations/partner-integrations.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+![Layer0 logo](./assets/Layer0.svg) | [Layer0](https://layer0.co/) extends the capabilities of a traditional CDN by not only hosting your static content, but also providing server-side rendering for progressive web applications. Layer0 allows caching both your APIs and HTML at the network edge to provide your users with the fastest browsing experience. Teams can ship faster leveraging an enhanced developer experience to deploy code faster and with more frequency, view their code quickly in atomically deployed environments, and integrating their CDN configuration to the overall build process. Layer0 provides the tools needed to build the modern apps capable of providing the performance expected by modern consumers.  | [Edge Optimization](https://github.com/aws-samples/retail-demo-store/blob/master/workshop/8-PerformanceOptimization/8.1-Layer0.ipynb) | 200 | 1 hour
```
Comment (gfaires) [inline docs/partner-integrations/partner-integrations.md:10]: Should add an indicator to highlight that this is currently going through a refresh and unlikely to work

File: docs/Deployment/getting-started.md [hunk 3]
```diff
@@ -1,26 +1,87 @@
...
+```
+You need Python 3.12.3 or higher
+If not: ```sudo apt install python3```
+
+## Install Git and clone repo:
+```
+sudo apt install git
+mkdir RetailDemoStore
+cd RetailDemoStore/
+git init
```
[the remark this answers is not in the document]
Comment (BastLeblanc) [reply docs/Deployment/getting-started.md:41]: it is not, well spotted

File: docs/Technical Guides/EnablingShopperProfiles.md (hunk 1 above)
Comment (BastLeblanc) [reply docs/Technical Guides/EnablingShopperProfiles.md:1]: This is correct, I haven't found a fully automated way to do it. I see options:

- Remove the demo guide from inside the demo itself (and replace with a link to this documentation in the header or with an iframe)
- Automate the demo guide generation into 
  - the vue components
  - this mkdocs doc
- Don't include these docs in the mkdocs and only in the demo 

I prefer the first option but we would benefit from a bit of a UI refresh (a better header, remove/change the demo guide button, ...)

to be discussed?

File: docs/Deployment/local-development/0-local-development-instructions.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+> DynamoDB is one dependency that can be run locally rather than connecting to the real DynamoDB. This makes it easier to run services like the [products](https://github.com/aws-samples/retail-demo-store/tree/master/src/products) service completely local to you computer.
+
+```console
+foo@bar:~$ aws sts get-session-token
```
[the remark this answers is not in the document]
Comment (BastLeblanc) [reply docs/Deployment/local-development/0-local-development-instructions.md:41]: see previous comment, this section is outdated

File: docs/Deployment/local-development/0-local-development-instructions.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+Some services, such as the [products](https://github.com/aws-samples/retail-demo-store/tree/master/src/products) and [recommendations](https://github.com/aws-samples/retail-demo-store/tree/master/src/recommendations) services, need to access AWS services running in your AWS account from your local machine. Given the differences between these container setups, different approaches are needed to pass in the AWS credentials needed to make these connections. For example, for the recommendations service we can map your local `~./.aws` configuration directory into the container's `/root` directory so the AWS SDK in the container can pick up the credentials it needs. Alternatively, since the products service is packaged from a [scratch image](https://hub.docker.com/_/scratch), credentials must be passed using the `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, and `AWS_SESSION_TOKEN` environment variables. In this case, rather than setting these variables in `.env` and risk exposing these values, consider setting these three variables in your shell environment. The following command can be used to obtain a session token which can be used to set your environment variables in your shell.
```
[the remark this answers is not in the document]
Comment (BastLeblanc) [reply docs/Deployment/local-development/0-local-development-instructions.md:36]: removing most of this part, now the services just pick up the current aws credentials

**Review by gfaires — APPROVED** [review]
LGTM

**Pull request merged**